core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1270,6 +1267,50 @@ class SnappySession(@transient private val sc: SparkContext,
     }
   }
 
+  def invalidateAll(): Unit = {
+    sessionCatalog.invalidateAll()
+  }
+
+  private[sql] def alterTable(tableName: String, isAddColumn: Boolean,
+                 column: StructField): Unit = {
+    alterTable(sessionCatalog.newQualifiedTableName(tableName), isAddColumn, column)
+  }
+
+  private[sql]  def alterTable(tableIdent: QualifiedTableName, isAddColumn: Boolean,
+                              column: StructField): Unit = {
+    val plan = try {
+      sessionCatalog.lookupRelation(tableIdent)
+    } catch {
+      case tnfe: TableNotFoundException =>
+        throw tnfe
+    }
+    plan match {
+      case LogicalRelation(c: ColumnFormatRelation, _, _) =>
```

Comment:
User should not even tru for ExternalTables. Should throw analysisexception in that case also. Also temp tables handled below should throw exception.